cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -614,6 +623,114 @@ func (s *Server) handleHTTPRoot(w http.ResponseWriter, r *http.Request) {
 	_ = repoTmpl.Execute(w, data)
 }
 
+func (s *Server) handleDebugList(w http.ResponseWriter, r *http.Request) {
+	withIndexed := true
+	if b, err := strconv.ParseBool(r.URL.Query().Get("indexed")); err == nil {
+		withIndexed = b
+	}
+
+	var indexed []uint32
+	if withIndexed {
+		indexed = listIndexed(s.IndexDir)
+	}
+
+	repos, err := s.Sourcegraph.List(r.Context(), indexed)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	bw := bytes.Buffer{}
+
+	tw := tabwriter.NewWriter(&bw, 16, 8, 4, ' ', 0)
+
+	_, err = fmt.Fprintf(tw, "ID\tName\n")
+	if err != nil {
+		http.Error(w, fmt.Sprintf("writing column headers: %s", err), http.StatusInternalServerError)
+		return
+	}
+
+	s.queue.mu.Lock()
+	name := ""
+	for _, id := range repos.IDs {
+		if item := s.queue.get(id); item != nil {
+			name = item.opts.Name
+		} else {
+			name = ""
+		}
+		_, err = fmt.Fprintf(tw, "%d\t%s\n", id, name)
+		if err != nil {
+			debug.Printf("handleDebugList: %s\n", err.Error())
+		}
+	}
+	defer s.queue.mu.Unlock()
```

Comment:
this is a defer. I think you meant to unlock right away :P
```suggestion
	s.queue.mu.Unlock()
```